### PR TITLE
Store and reuse cached compiled assets

### PR DESCRIPTION
**Pull request by genevieve**
Acceptance Criteria:
1. Build rails image
1. Build again without any changes, verify `Reusing cache layer` - check that assets are in image at `public/assets` and `tmp/cache/assets` that are directories linked to the layer
1. Change asset in `app/assets`
1. Verify re-build - check that assets are in image at `public/assets` and `tmp/cache/assets` that are directories linked to the layer

### CODE REVIEW

**Comment by genevieve** [comment]
@sophiewigmore Do you want to try running the tests from this branch instead?

**Comment by genevieve** [comment]
First run:

```
Paketo Rails Assets Buildpack 0.0.2
  Executing build process
    Running 'bundle exec rails assets:precompile assets:clean'
      Completed in 3.735s
```

Second run:

```
Paketo Rails Assets Buildpack 0.0.2
  Reusing cached layer /layers/paketo-buildpacks_rails-assets/assets
```

After assets change:
```
Paketo Rails Assets Buildpack 0.0.2
  Executing build process
    Running 'bundle exec rails assets:precompile assets:clean'
      Completed in 3.897s
```

Fourth run:
```
Paketo Rails Assets Buildpack 0.0.2
  Reusing cached layer /layers/paketo-buildpacks_rails-assets/assets
```

File: build.go [hunk 1]
```diff
@@ -1,26 +1,96 @@
...
+			if !os.IsNotExist(err) {
+				return packit.BuildResult{}, fmt.Errorf("failed to stat %s: %w", layerTmpAssetsCache, err)
+			}
+		} else {
+			err = os.Symlink(filepath.Join(layerTmpAssetsCache), filepath.Join(context.WorkingDir, tmpAssetsCacheDir))
+			if err != nil {
+				return packit.BuildResult{}, fmt.Errorf("failed to symlink tmp/assets/cache into working directory: %w", err)
+			}
+		}
+
```
Comment (ryanmoran) [inline build.go:93]: I'm not sure that you need to check to see if the target locations exist before creating the symlink. Should you just create the symlink always?

File: build.go [hunk 2]
```diff
@@ -2,46 +2,107 @@ package railsassets
...
-		duration, err := clock.Measure(func() error {
-			return buildProcess.Execute(context.WorkingDir)
-		})
+		appAssetsDir := filepath.Join(context.WorkingDir, "app", "assets")
+		sum, err := calculator.Sum(appAssetsDir)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		err = environmentSetup.ResetLocal(context.WorkingDir)
```
Comment (sophiewigmore) [inline build.go:54]: Why do you need to clear out the working directory on each build?

**Comment by sophiewigmore** [comment]
This looks pretty good to me!

- I like the use of the environment setup methods.

- When I rebuild an app, I see that the `rails-assets` logs `Reusing cached layer /layers/paketo-buildpacks_rails-assets/assets`, and the contents of the layers directory is populated.
- Additionally, all of the unit tests and integration tests pass for me now

**Pull request merged**